python/mxnet/gluon/data/dataset.py

```diff
@@ -173,8 +173,15 @@ class RecordFileDataset(Dataset):
         Path to rec file.
     """
     def __init__(self, filename):
-        idx_file = os.path.splitext(filename)[0] + '.idx'
-        self._record = recordio.MXIndexedRecordIO(idx_file, filename, 'r')
+        self._filename = filename
+        self.reload_recordfile()
+
+    def reload_recordfile(self):
+        """
+        Reload the record file.
+        """
+        idx_file = os.path.splitext(self._filename)[0] + '.idx'
+        self._record = recordio.MXIndexedRecordIO(idx_file, self._filename, 'r')
```

Comment:
we should fix this in backend. Why would forking cause a problem? File descriptors should be duplicated when forking

Reply:
That would be the ideal solution indeed.

Reply:
isn't close_fds false by default?

Reply:
For the subprocess package, it is `True` by default.

Reply:
Ok digging a bit more, it seems that the `multiprocessing` package does not close file descriptors since it is simply calling `os.fork()`. I have updated the description of the PR to reflect the issue. tldr: file descriptors keep track of the position they are in the file. When forking, all process refer to the same file descriptor and try to move the current position at the same time, causing a crash.